### ISSUE TITLE
nf-core download: customise config to include path to singularity image

### ISSUE THREAD

**Issue by ewels** [nf-core/tools#464]
We can run `nf-core download <pipeline> --singularity` to fetch the pipeline files and singularity image.

The code nicely edits the pipeline `nextflow.config` file so that the custom configs path is relative to the pipeline code for the institutional configs. However, this is not done for the singularity path, so the user still has to use `-with-singularity` to specify the location for these images.
It would be nice if the tool could edit the configs somehow so that the singularity image is automatically found with just `-profile singularity`.

Could potentially do this with `process.container`, or maybe be more clever & generic by setting the singularity cache directory path or something?

**Comment by MaxUlysse** [comment]
We could maybe play with `singularity.cacheDir`.
By default they are in `work/singularity`
We can change that, or download the singularity images there.

Who is using nf-core download to get singularity images?

**Comment by ewels** [comment]
We will hopefully use it for irma soon :)

**Comment by MaxUlysse** [comment]
Yes, but then we don't really need to download the singularity images with `nf-core download`, but we still would want to set up the path

**Comment by ewels** [comment]
Just need to update this function to also edit `process.container` I guess? Or something similar..

https://github.com/nf-core/tools/blob/ebf63095ab222e4df987c4e950ce3839e8784d3e/nf_core/download.py#L226-L243

**Comment by ewels** [comment]
Ok, after a bit of testing trial and error, this little gem seems to do the job:

```nextflow
singularity.cacheDir = "${projectDir}/../singularity-images/"
```

I think we're on to a winner 🎉

**Comment by ewels** [comment]
Implemented in the upcoming v1.13 release of tools. See #832.
